components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -165,7 +165,7 @@ public void handle(List<String> roles, String subject, Map<String, String> attri
                 }
             }
 
-            if (roles != null && !roles.isEmpty()) {
+            if (roles != null) {
```

Comment:
A little bit confused about what's happening here. Though we check `roles != null`, `roles` variable is not considered within the `if` block.

Reply:
This `roles` is used in the above lines to create the list of roles to add list, then that list is used inside this IF block to update the user's new role set. If the `roles` is null we don't update the user's roles.

Though the order seems a bit confusing `!roles.isEmpty()` check always become false if we have only user role and later remove that role.